File: gradle-nonnull-plugin/sample/app/src/main/java/com/novoda/gradle/nonnull/MainActivity.java
package com.novoda.gradle.nonnull;

import android.app.Activity;
import android.os.Bundle;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // this should result in a warning in the IDEA
        setValue(null);
    }

    private void setValue(Object value){
        // no-op
    }
}

File: AndroidKotlinALL4MPV/app/src/main/java/novoda/androidkotlinall4mvp/MainActivity.kt
package novoda.androidkotlinall4mvp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val presenter: Presenter = TODO()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        presenter.foo(this, home_view::display)

        home_view.setEventReporter(presenter::reportEvent)
    }

}
